### DISCUSSION TITLE
Official workaround for not being able to select rows for Rest endpoints?

### DISCUSSION THREAD

**Discussion by SomniVertix** [Budibase/budibase#10354] [Help]
Is there an official workaround for trying to select a row when the data source is an endpoint? I know its not generally supported, but Im looking for any workaround that I can try

**Comment by melohagan** [comment]
Hey @SomniVertix 

The reason selecting rows for a REST API response is that there is no way to uniquely identify the rows -> there's no **_id** fields.

I wouldn't say there's an official workaround, however the approach I would go for with this one would be to add a repeater.

Then you could have an icon button with conditional UI to emulate a checkbox. Then on click use **App State** to store the 'checked' state of the row. You'll need to determine which fields of the row uniquely identify the row.

Docs:
 - https://docs.budibase.com/docs/app-state
 - https://docs.budibase.com/docs/conditions
 - https://docs.budibase.com/docs/repeater
 
You can also have a look at this answer to see how to use **App State** as a dictionary, although you'll be storing the state of a repeater row, and not a form field as shown:
https://github.com/Budibase/budibase/discussions/9536#discussioncomment-4960342

**Reply by SomniVertix** [reply]
Would adding an _id field to the API response circumvent this possibly? if not, that's fine, I can look into the repeater scenario

**Reply by melohagan** [reply]
Hey @SomniVertix an **_id** field that's added to the API response will not be treated as a unique identifier automatically in Budibase.

You will have to go down the Repeater route!

**Reply by SomniVertix** [reply]
Gotcha - I also wonder: Is it possible to have a REST Endpoint run on page load, dump all of the data into the budibase DB, and then display that as the fields? So that you get all the features from using the in-house DB?

**Reply by micebrain** [reply]
> Hey @SomniVertix an **_id** field that's added to the API response will not be treated as a unique identifier automatically in Budibase.
> 
> You will have to go down the Repeater route!

Hello, it would be nice to have the unique identifier field configurable, with the default being the internal `_id`. This would be a useful enhancement to have and make it possible to connect REST API data source and potentially other data sources.

**Reply by melohagan** [reply]
You can connect them with a "soft link", which is simply adding a filter on one datasource that matches on the field from a different datasource
